File: app/src/main/java/org/autojs/autojs/util/MathUtils.java
package org.autojs.autojs.util;

/**
 * Created by Stardust on Nov 26, 2017.
 */
public class MathUtils {

    public static int min(int... ints) {
        int min = ints[0];
        for (int i = 1; i < ints.length; i++) {
            min = Math.min(ints[i], min);
        }
        return min;
    }
}

File: app/src/main/java/org/autojs/autojs/util/MathUtils.kt
package org.autojs.autojs.util

/**
 * Created by Stardust on Nov 26, 2017.
 * Transformed by SuperMonster003 on Mar 5, 2026.
 * Modified by SuperMonster003 as of Mar 5, 2026.
 */
object MathUtils {

    @JvmStatic
    fun min(vararg ints: Int): Int =
        ints.minOrNull() ?: throw NoSuchElementException("Array is empty")

    @JvmStatic
    fun Int.floorMod(mod: Int): Int =
        Math.floorMod(this, mod)

    @JvmStatic
    fun Long.floorMod(mod: Long): Long =
        Math.floorMod(this, mod)

    @JvmStatic
    fun Long.floorMod(mod: Int): Int =
        Math.floorMod(this, mod)
}
